MusicNotationKit/MusicNotationKit/NoteCollection.swift

```diff
@@ -25,6 +25,9 @@ public protocol NoteCollection {
      */
     var noteTimingCount: Int { get }
 
+    var first: Note { get }
```

Comment:
We should probably add documentation to `first` and `last` mentioning that the library should die if `NoteCollection` is empty.  Is throwing `NoteCollectionError.Empty` an overkill right now?